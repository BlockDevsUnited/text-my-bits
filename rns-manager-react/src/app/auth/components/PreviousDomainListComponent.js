import React from 'react';
import propTypes from 'prop-types';

const PreviousDomainListComponent = ({ previousDomains, switchLoginClick }) => (
  <div className="switch">
    {previousDomains.map(addr => (
      <button
        key={addr.domain}
        type="button"
        className="switchButton"
        onClick={() => switchLoginClick(addr.domain)}
      >
        {addr.domain}
      </button>
    ))}
  </div>
);

PreviousDomainListComponent.propTypes = {
  previousDomains: propTypes.arrayOf(propTypes.shape({
    domain: propTypes.string,
    owner: propTypes.string,
  })).isRequired,

  switchLoginClick: propTypes.func.isRequired,
};

export default PreviousDomainListComponent;
