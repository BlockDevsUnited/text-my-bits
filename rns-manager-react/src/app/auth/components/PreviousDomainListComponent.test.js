import React from 'react';
import { shallow } from 'enzyme';
import PreviousDomainListComponent from './PreviousDomainListComponent';

describe('PreviousDomainListComponent', () => {
  const sampleDomains = [
    { domain: 'foobar.rsk', owner: '0x123456' },
    { domain: 'barfoo.rsk', owner: '0x123456' },
  ];

  it('renders without crashing', () => {
    const component = shallow(
      <PreviousDomainListComponent previousDomains={sampleDomains} switchLoginClick={jest.fn()} />,
    );
    expect(component).toMatchSnapshot();

    expect(component.find('button').at(0).text()).toBe('foobar.rsk');
    expect(component.find('button').at(1).text()).toBe('barfoo.rsk');
  });

  it('handles interaction', () => {
    const switchLoginClick = jest.fn();

    const component = shallow(
      <PreviousDomainListComponent
        previousDomains={sampleDomains}
        switchLoginClick={switchLoginClick}
      />,
    );

    component.find('button').at(0).simulate('click');
    expect(switchLoginClick).toBeCalledWith('foobar.rsk');

    component.find('button').at(1).simulate('click');
    expect(switchLoginClick).toBeCalledWith('barfoo.rsk');
  });
});
