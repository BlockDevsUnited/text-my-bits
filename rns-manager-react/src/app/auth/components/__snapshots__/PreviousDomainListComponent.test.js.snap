// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviousDomainListComponent renders without crashing 1`] = `
<div
  className="switch"
>
  <button
    className="switchButton"
    key="foobar.rsk"
    onClick={[Function]}
    type="button"
  >
    foobar.rsk
  </button>
  <button
    className="switchButton"
    key="barfoo.rsk"
    onClick={[Function]}
    type="button"
  >
    barfoo.rsk
  </button>
</div>
`;
